### PR TITLE
Deletion of Byte Collection and Collection Category | UI changes | Se…

### PR DESCRIPTION
**Pull request by UsmanSaleem2003**
…lection of Single Byte one time in Byte Collection

# Pull Request Template

Before submitting this PR, please make sure you have completed the following checklist. This helps ensure that our codebase remains high quality and that our features work well across all supported devices and themes.

### Code Review
- [x] I have carefully reviewed my code and believe it is final from my side.
- [x] I have ensured there is no hardcoded color in my changes. Colors should be defined in theme files and reused.

### Theme Compatibility
- [] I have tested the new feature or updates in both light and dark themes.
- [] I confirm that the visual aspects of the feature are compatible and appealing in both themes.

### Device Compatibility
- [x] I have tested my changes on mobile screen sizes.
- [x] I have tested my changes on tablet screen sizes.
- [x] I have tested my changes on laptop screen sizes.

### Functional Testing
- [x] I have thoroughly tested the functionality of my changes.
- [x] I am confident that my changes do not negatively impact existing features.

### Testing Documentation
Please provide a brief description of the tests you have performed to verify your changes. Include any relevant scenarios or edge cases.

- Example test 1: Tested feature X by uploading the image
- Example test 2: Tested feature X by doing these things.

# Demo Video
Please provide a demo video of your changes. This helps reviewers understand the changes and verify the functionality.

You can record the video using tools like Loom(https://www.loom.com/), Clip by ClickUp(https://clickup.com/features/clips), Sendspark(https://www.sendspark.com/), etc.

Include the link to the video below and make sure the video covers
- [x] Explain the task/bug you are solving
- [x] Show the changes in the UI
- [x] Show the changes in the codebase
- [x] Show the changes you tested to make sure it works
- [x] Show the changes in various screen sizes

**Loom Video Link :**  https://www.loom.com/share/41a791766ffe4781a5684ba96a38e2ba?sid=b1aaf8de-37bf-43e5-8049-d61798347c6c

**Major Changes :**
Added Deletion Functionality in byteCollection and byteCollectionCategory
Corrected Selection of More than one byte in ByteCollection
Corrected Labels and added * for required fields

### CODE REVIEW

**Review by MSamiTariq — APPROVED** [review]
Looks good to me

**Comment by UsmanSaleem2003** [comment]
**Loom video link :** https://www.loom.com/share/b20eb4749db14b5ba1457214022f2723?sid=c6670af5-45a7-4f0c-852d-789de57fd61a

I have created a new commit in the same PR, in which I have : 

1. corrected the deletion logic of collection in collection-category.
2. corrected the selection of collection in collection-category

File: academy-ui/src/components/byteCollectionCategory/ByteCollectionCategoryEditor.tsx [hunk 1]
```diff
@@ -138,11 +138,13 @@ function ByteCollectionCategoryEditor(props: ByteCollectionCategoryEditorProps)
               onClose={() => setShowSelectByteCollectionModal(false)}
               addByteCollection={(byteCollections: ByteCollectionFragment[]) => {
                 byteCollections.forEach((byteColl) => {
-                  helperFunctions.addByteCollection(byteColl);
+                  if (!byteCategory.byteCollections.some((existingByte) => existingByte.id === byteColl.id)) {
+                    helperFunctions.addByteCollection(byteColl);
+                  }
```
Comment (MSamiTariq) [inline academy-ui/src/components/byteCollectionCategory/ByteCollectionCategoryEditor.tsx:143]: what I can deduce from this logic is that when to open the modal to add byte collections, the collections which are already added will also be shown, just that you have added a check to not add it again, am I right?

if that's the case, we need to change the logic, we shouldn't be displaying the already selected collections on the selection modal

Comment (MSamiTariq) [reply academy-ui/src/components/byteCollectionCategory/ByteCollectionCategoryEditor.tsx:143]: BTW this deduplication logic is already there in helperFunctions.addByteCollection

File: academy-ui/src/graphql/byte/byteCollectionCategory.graphql [hunk 2]
```diff
@@ -19,6 +20,7 @@ fragment CategoryWithByteCollection on CategoryWithByteCollection {
   byteCollections {
     ...ByteCollection
   }
+  archive
```
Comment (MSamiTariq) [inline academy-ui/src/graphql/byte/byteCollectionCategory.graphql:23]: move this field above the byteCollections object

File: academy-ui/src/components/byteCollectionCategory/SelectByteCollectionModal.tsx [hunk 3]
```diff
@@ -6,69 +6,79 @@ import FullPageModal from '@dodao/web-core/components/core/modals/FullPageModal'
...
+  const handleCardClick = (byteCollection: ByteCollectionFragment) => {
+    if (tempSelectedByteIds.includes(byteCollection.id)) {
+      setTempSelectedByteIds(tempSelectedByteIds.filter((id) => id !== byteCollection.id));
+      setTempSelectedByteCollections(tempSelectedByteCollections.filter((collection) => collection.id !== byteCollection.id));
+    } else {
+      setTempSelectedByteIds([...tempSelectedByteIds, byteCollection.id]);
+      setTempSelectedByteCollections([...tempSelectedByteCollections, byteCollection]);
+    }
+  };
+
```
Comment (MSamiTariq) [inline academy-ui/src/components/byteCollectionCategory/SelectByteCollectionModal.tsx:45]: can you explain this logic? no idea what's going on in here

File: academy-ui/src/components/byteCollectionCategory/ByteCollectionCategoryEditor.tsx (hunk 1 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollectionCategory/ByteCollectionCategoryEditor.tsx:143]: No, once the selected byte collections are added, they will not be shown in the card from where you selected them last time. 

Moreover, for the case of duplication, I can remove that if condition and keep it simple. Just let me know, if you want that.

File: academy-ui/src/components/byteCollectionCategory/SelectByteCollectionModal.tsx (hunk 3 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/components/byteCollectionCategory/SelectByteCollectionModal.tsx:45]: yes sure,
The logic is simple and that is, on handleCardClick, if the tempSelectedByteIds have the id of byteCollection, then :

1. The ID is removed from tempSelectedByteIds.
2. The corresponding byte collection object is removed from tempSelectedByteCollections.

Now, in else condition, (if the tempSelectedByteIds don't have the byteCollection.id ) then simply, keep the previous ones (can be empty at start) using spread operator and just add the clicked byteCollection.


 _const availableByteCollections =
    byteCollectionsResponse?.byteCollections.filter((byteCollection) => !byteCollectionSummaries.some((summary) => summary.id === byteCollection.id)) || [];_
    
for above code, it's applying filtering function on byteCollections from byteCollectionsResponse to check if any byteCollection's id matches the id of byteCollectionSummaries
The purpose of that function is, to exclude any collection id which is already in the Summaries

File: academy-ui/src/graphql/byte/byteCollectionCategory.graphql (hunk 2 above)
Comment (UsmanSaleem2003) [reply academy-ui/src/graphql/byte/byteCollectionCategory.graphql:23]: Done

File: academy-ui/src/components/byteCollectionCategory/ByteCollectionCategoryEditor.tsx (hunk 1 above)
Comment (MSamiTariq) [reply academy-ui/src/components/byteCollectionCategory/ByteCollectionCategoryEditor.tsx:143]: yes, that condition is already present, moreover if an already selected byte collection won't be shown, then I think we don't need this condition, so you can remove this

**Review by MSamiTariq — APPROVED** [review]
Approving the changes, but make sure to correct the responsiveness in the following PR

**Pull request merged**